pkg/cluster/deploybaseresources_additional.go

```diff
@@ -20,7 +20,23 @@ import (
 )
 
 func (m *manager) denyAssignment() *arm.Resource {
-	return &arm.Resource{
+	excludePrincipals := []mgmtauthorization.Principal{}
+	if m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile != nil {
+		excludePrincipals = append(excludePrincipals, mgmtauthorization.Principal{
+			ID:   &m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.SPObjectID,
+			Type: to.StringPtr("ServicePrincipal"),
+		})
+	}
+	if m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile != nil {
+		for _, identity := range m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities {
+			excludePrincipals = append(excludePrincipals, mgmtauthorization.Principal{
+				ID:   &identity.ObjectID,
+				Type: to.StringPtr(string(mgmtauthorization.ServicePrincipal)),
```

Comment:
Is there reasoning behind having the `Type` here different than the one we have on line 27?

Reply:
My thinking was that the `mgmtauthorization.Principal` for the `ServicePrincipal` would stay the same (aside from indentation) since this patch isn't really touching it. But since the `mgmtauthorization.Principal`s for identities were net new, I would reference the `mgmtauthorization.PrincipalType` const. Pretty much just a "I'm not really touching this part other than moving it" thing

It's a pretty fine hair to split. I can match them up if you'd like

Reply:
IMO, it is better to get them matched. I am not sure though which one's to use `mgmtauthorization.ServicePrincipal` or `ServicePrincipal`.

Reply:
I've standardized on `mgmtauthorization.ServicePrincipal` in the amended PR